CoffeeChat/Cells/OnboardingTableViewCell.swift

```diff
@@ -63,23 +68,49 @@ class InterestsTableViewCell: UITableViewCell {
 
         titleLabel.snp.makeConstraints { make in
             make.leading.equalTo(interestImageView.snp.trailing).offset(textSidePadding)
-            make.top.equalTo(cellBackgroundView).inset(sidePadding)
+            if showingInterests {
+                make.top.equalTo(cellBackgroundView).inset(sidePadding)
+            } else {
+                make.centerY.equalToSuperview()
+            }
         }
 
-        categoriesLabel.snp.makeConstraints { make in
-            make.leading.equalTo(titleLabel)
-            make.top.equalTo(titleLabel.snp.bottom)
+        if showingInterests {
+            categoriesLabel.snp.makeConstraints { make in
+                make.leading.equalTo(titleLabel)
+                make.top.equalTo(titleLabel.snp.bottom)
+            }
         }
     }
 
     func configure(with interest: Interest) {
         titleLabel.text = interest.name
         categoriesLabel.text = interest.categories
+        // Determine if a relayout is needed (was showing Group and configured with an Interest)
+        if !initialized || !showingInterests {
+            relayoutSubviews(interests: true)
+        }
+    }
+
+    func configure(with group: Group) {
+        titleLabel.text = group.name
+        // Determine if a relayout is needed (was showing Interest and configured with an Group)
+        if !initialized || showingInterests {
+            relayoutSubviews(interests: false)
+        }
+    }
+
+    /// Removes and resets constraints on the cell. Includes categoriesLabel in the layout if `interests` is true
+    private func relayoutSubviews(interests: Bool) {
+        showingInterests = interests
+        initialized = true
+        cellBackgroundView.subviews.forEach { $0.removeConstraints($0.constraints)}
```

Comment:
instead of doing this, you could also just change your `setupConstraints` to use `remakeConstraints` i think